x-pack/plugins/ml/public/application/jobs/new_job/common/job_creator/job_creator.ts

```diff
@@ -241,7 +238,7 @@ export class JobCreator {
   public set jobId(jobId: JobId) {
     this._job_config.job_id = jobId;
     this._datafeed_config.job_id = jobId;
-    this._datafeed_config.datafeed_id = `datafeed-${jobId}`;
+    this._datafeed_config.datafeed_id = createDatafeedId(jobId);
```

Comment:
Not related to the changes here as also happens on `main`, but hit this error when creating a job off the Patterns tab in Discover and then switching to the job wizard:

<img width="1127" alt="Screenshot 2024-09-09 at 10 17 42" src="https://github.com/user-attachments/assets/9b13f0e0-7dac-438b-a422-02a8462df7e5">

<img width="1703" alt="Screenshot 2024-09-09 at 10 35 58" src="https://github.com/user-attachments/assets/0b02cff2-a2f2-4073-a2b8-44f9c3049d79">

Reply:
I've raised a separate [PR](https://github.com/elastic/kibana/pull/192375) to fix this